src/modules/mavlink/mavlink_timesync.cpp

```diff
@@ -86,11 +86,14 @@ MavlinkTimesync::handle_message(const mavlink_message_t *msg)
 			timespec tv = {};
 			px4_clock_gettime(CLOCK_REALTIME, &tv);
 
-			// date -d @1234567890: Sat Feb 14 02:31:30 MSK 2009
-			bool onb_unix_valid = (unsigned long long)tv.tv_sec > PX4_EPOCH_SECS;
-			bool ofb_unix_valid = time.time_unix_usec > PX4_EPOCH_SECS * 1000000ULL;
-
-			if (!onb_unix_valid && ofb_unix_valid) {
+			// Set the system time if we are lagging behind by more than a minute
+			hrt_abstime seconds_behind = 60;
+			hrt_abstime local_time = tv.tv_sec;
```

Comment:
I find it confusing to use `hrt_abstime` for seconds, as it's usually microseconds. I'd probably just use `uint64_t`. I know it's the same under the hood but it did trip me up reading it.

Reply:
I changed it to time_t which has underlying type uint32_t. This matches tv_sec and the px4_epoch definition. But this limits us to only 137 years :fearful: